GitUI/CommandsDialogs/BrowseDialog/GitStatusMonitor.cs

```diff
@@ -256,7 +256,7 @@ void GitUICommands_PostCheckout(object sender, GitUIPostActionEventArgs e)
         private void StartWatchingChanges(string workTreePath, string gitDirPath)
         {
             // reset status info, it was outdated
-            GitWorkingDirectoryStatusChanged?.Invoke(this, new GitWorkingDirectoryStatusEventArgs());
+            GitWorkingDirectoryStatusChanged?.Invoke(this, null);
```

Comment:
I agree regarding the Event design guidelines and have reverted the `null` EventArgs.
I handled the null pointer in the code you mention. Now, the `?.` operator has been shifted one position to the right.

Reply:
Is there a particular reason why you insist on removing `ItemStatuses = Enumerable.Empty<GitItemStatus>();` from `GitWorkingDirectoryStatusEventArgs` ctor?
This way we now need to add null checks in other places...

Reply:
Yes, as Gerhard mentioned, too, it was incomplete. The now dedicated state "no information yet" shall not be equal to the state "nothing to commit".

Reply:
Would you prefer an empty list and an additional flag indicating the validity of the information?

Reply:
This should probably be changed to not create an event at all and let Browse handle changed repos.

Reply:
A null event could have been a solution too. This update caused some problems for the Submodule status update just merged (but I choose to solve that in a different way).

Reply:
@gerhardol could you please elaborate on problems you've encountered.

Reply:
> The now dedicated state "no information yet" shall not be equal to the state "nothing to commit".

Right, thank you for a clarification.
How would you solve it with a flag?

Reply:
> How would you solve it with a flag?

The event could have a flag `bool StateUnknown` or `Updating` and an empty list for those consumers that do not need to indicate the intermediate state.
Though I dislike this approach because code shall never rely on pointers not being null (unless it is ensured explicitely, e.g. by `[NotNull]`). I.e. each consumer must check for the null pointer. Then the additional flag would be redundant.